packages/synthetics-sdk-broken-links/src/broken_links.ts

```diff
@@ -82,31 +83,65 @@ export async function runBrokenLinks(
   let browser: Browser;
   try {
     const options = processOptions(inputOptions);
+    let timeLimitTimeout;
+    let timeLimitresolver = () => {};
+    const timeLimitPromise = new Promise<void>((resolve) => {
+      timeLimitresolver = () => {
+        resolve();
+      };
+      const time_used = Date.now() - new Date(startTime).getTime();
+      timeLimitTimeout = setTimeout(
```

Comment:
Upon reviewing this code, im wondering if we really need to manually resolve + clear the timeout after promise.race() happens. Is there a strong reason to do the manual cleanup, rather than just allowing it to self cleanup?